### DISCUSSION TITLE
OBS Studio 28.0 and you: What you need to know as a StreamFX user

### DISCUSSION THREAD

**Discussion by Xaymar** [Xaymar/obs-stream-effects#879] [Announcements]
As the release of OBS Studio 28.0(.0) is looming on the horizon, StreamFX has a lot of catching up to do. A lot of work happened over the last weeks to make sure that StreamFX can be used and built with the latest OBS Studio version, despite it not being out yet. Let's go over what exactly will change

#### OBS Studio 28.0 is now the minimum OBS version
Due to massive build system, API and dependency changes, it is not possible to support both 27.2 and 28.0 and beyond in a single binary. Therefore in order to make it easier on myself, the minimum OBS Studio version is now 28.0.

#### MacOS: ARM64 support on MacOS
As OBS Studio is also heading in this direction, StreamFX is following suit and opting to support ARM64 natively. The support for ARM64 requires a MacOS version of at least 11.0 or higher, so no 10.15 for the ARM64 crowd. As we're currently running into a bug with this support, these binaries will be available either with the full release of v0.12 or when the bug is fixed.

#### System Requirements are now on each Release
With the requirements changing from version to version, it really doesn't make sense to put these in the documentation. Instead from now on they'll be at the bottom of the Release Notes, formatted to be easily readable by anyone. This change is not retroactive, and was only applied to the latest stable and current alpha/beta versions.

![image](https://user-images.githubusercontent.com/437395/185539324-13c4c53e-d78e-4d5f-86ef-5c31f3718afa.png)

#### Deprecation of the AMD AMF (via FFmpeg) Encoder
With OBS Studio 28.0 the old obs-amd-encoder project is finally laid to rest, and a new AMD AMF integration is provided. This AMD AMF integration is technically and functionally superior, especially due to the FFmpeg AMD AMF encoder practically being unmaintained. Therefore users should prepare to migrate to the new OBS Studio integrated Encoder when OBS Studio 28.0 is released.

#### Deprecation of the AV1 (via AOM) Encoder
Same story as above, the encoder is no longer necessary. The integrated encoder solves most if not all situations, and for the remaining situations you have the other integrated encoders. And if those aren't enough, there's also ProRES and DNxHR.

#### No more Auto-Dependency system
Originally this system was extremely helpful, working in every situation, easily able to handle most if not all problems thrown at it. As time went on, more and more problems occured for which solutions were non-existent - if you ran into the problem(s) you were just screwed. So as I (and probably others) got extremely tired of this system doing its best to be the worst in the world, it was carefully removed from reality, never to bother developers again.

#### Formatting validation is now functional on all Platforms
Mainly affects people writing PRs, as without passing this test your PR can't be merged. Run clang-format before pushing!

**Comment by hannesmann** [comment]
How long is the AMF FFmpeg encoder going to be kept around? I use AMF on Linux because it has better performance and quality compared to VAAPI in OBS. The new AMF integration in 28.0 isn't available on Linux, so removing the StreamFX option leaves Linux users without any AMF support until #861 is resolved.

**Reply by Xaymar** [reply]
It is deprecated with v0.12, removing it from the usable encoders. Once deprecated, no further guarantees for functionality, stability, or it existing is made. The reason is as stated, the FFmpeg integration of AMD AMF is unmaintained so it is horribly out of date. With OBS 28.0, it effectively no longer works as users would expect.

**Reply by Xaymar** [reply]
There is a PR upstream that adds support for AMF on Linux: https://github.com/obsproject/obs-studio/pull/7206/files#diff-f9caf747e939c6360e5b5c298c08a81fdc21632a460ee8d82af38cdea923bf9c

**Comment by ZTHCreates** [comment]
So far no matter what i have done i haven't been able to get StreamFX to work with OBS 28.

**Reply by Xaymar** [reply]
There is an Alpha version out that supports OBS Studio 28.0.0. As a lot has changed, that is the only version able to load with OBS Studio 28.0.0 for now.

**Comment by jcoconnor** [comment]
OBS Studio 28.0 seems to have just dropped - any ETA on when StreamFX is going to be available ?

**Reply by Xaymar** [reply]
When it's done.

**Reply by RSMediaAU** [reply]
https://obsproject.com/kb/obs-studio-28-plugin-compatibility is showing it as compatible with Windows, however I'm unfortunately not able to load StreamFX. Should I downgrade OBS to 27.2.4 for now?

**Reply by Xaymar** [reply]
There are alpha builds that work with 28.0 if you can't wait to upgrade OBS. Unfortunately I don't have any control over what a third party claims to be true.

**Comment by marlop352** [comment]
Is there a list of what needs to be completed before v0.12 is considered done?

**Reply by Xaymar** [reply]
Public list? No. It will be done when it is done, and no earlier than then.

**Reply by marlop352** [reply]
That's a rather "interesting" stance, but ok.

**Reply by Xaymar** [reply]
It is a 1-person project after all, with hardly any funding. Most things are available in either the Discord or the Issue Tracker. As you can't get an average end-user to use GitHub correctly at all, you'll have to piece together information from both Discord and GitHub. And sometimes you'll also have to scour through the OBS Forums and OBS Discord to find a StreamFX bug that was incorrectly reported to OBS Studio.
